oauth2_http/java/com/google/auth/oauth2/DefaultCredentialsProvider.java

```diff
@@ -210,6 +217,13 @@ private final GoogleCredentials getDefaultCredentialsUnsynchronized(
     if (credentials == null) {
       LOGGER.log(Level.FINE, "Attempting to load credentials from GCE");
       credentials = tryGetComputeCredentials(transportFactory);
+      // tryGetComputeCredentials can return a null value. This check won't set the source
+      // if the ComputeEngineCredentials fails (prevents a NPE)
+      if (credentials != null) {
+        credentials =
+            credentials.withSource(
+                String.format("Metadata Server URL set to %s", getEnv(GCE_METADATA_HOST_ENV_VAR)));
```

Comment:
question: is `GCE_METADATA_HOST_ENV_VAR` always available? should we account for a default?

Reply:
can we reuse this  ComputeEngineCredentials method?
https://github.com/googleapis/google-auth-library-java/blob/5768f755e6091b8f52e8f3f16fd258f9ea4e9c89/oauth2_http/java/com/google/auth/oauth2/ComputeEngineCredentials.java#L603

Reply:
Good catch! Updating. Thanks!